### ISSUE TITLE
Port test_overquota

### ISSUE THREAD

**Issue by pjenvey** [mozilla-services/syncstorage-rs#120]
Port the equivalent of the check_storage_quota decorator: if a quota is set, ensure writes won't exceed it.

(This doesn't appear to have tests in test_storage.py)

**Comment by pjenvey** [comment]
There is a test_overquota though it doesn't run by default. The test suite sets `self.distant` to True which causes the test to skip

**Comment by pjenvey** [comment]
Since test_overquota doesn't run by default we should likely port it over to rust (likely as a test in server/test.rs) to replicate what it does.

**Comment by fzzzy** [comment]
There's already a rust test that covers this, test_quota, that's landed on master.

**Comment by pjenvey** [comment]
Reopening as test_quota only covers the db layer w/ a unit test.

The e2e test_overquota is worth porting as it covers write operation responses, e.g. that an `X-Weave-Quota-Remaining` header is returned along with a `quota-exceeded` status body.

I don't think we're doing either of these yet.

**Comment by tublitzed** [comment]
@fzzzy - heads up I just slotted this back into the queue. Let's knock this out so we can wrap up the quota work.

**Comment by fzzzy** [comment]
The rust server does not currently implement X-Weave-Quota-Remaining so we can't test it, but we also think the client never does anything with it. If the client never does anything with it, we may be safe to leave it out. Regardless, we should add a new bug for implementing it and testing it if we decide we do need it.

**Comment by fzzzy** [comment]
There's nothing to do here. The rust server does not implement X-Weave-Quota-Remaining so we can't test it, and db/tests/db.rs test_quota already tests is_err in case of an overquota situation. All this test would cover is whether Actix can turn this Err into a 302.

However, we don't have the appropriate test harness on the http side to change the Settings object to lower the quota to test the 302 situation. We could spend a lot of time rewriting the http test harness such that the Settings object can be passed through, but it touches every http test.

Given that test_quota already tests that the overquota and the test harness for changing the Settings object already exists in the db tests, this test would add less than one line of additional test coverage.

This test would best be implemented in a Python 3 integration test.

**Comment by fzzzy** [comment]
wip is here https://github.com/mozilla-services/syncstorage-rs/compare/overquota-test

**Comment by pjenvey** [comment]
This can't really be a Python e2e test easily as it requires changing syncstorage-rs's quota settings.

I've made `init_app!`'s settings configurable now, so we should be able to finish this up easily. A couple things I noticed:

- The Python test for "status" == "quota-exceeded" is wrong (it's using what I'd call "new style" responses, which were quickly reverted back to the original simple numeric response codes in the Python codebase). It should check the body for solely the OverQuota numeric error code (like other tests in the file look for "0" codes)
- I doubt the client does much with the header but let's ping the client team about it. Even if they don't currently do anything with it, it's possible they might want to now that we're actually enabling the quota

Even though it's a really basic test I think it's important to have something in place since before this goes to prod. It could be very handy later if we find new issues around quota support and need to add to the test or use it as a base for new tests.
